### PR TITLE
ensure jenkins works with plugins

### PR DESCRIPTION
**Pull request by joshrwolf**
I'm about the furthest thing from a jenkins knower, but apparently plugins are important.

This is ensure the scripts/dependencies needed for initializing, supporting, and downloading plugins are available in the image.

Most importantly, it ensures we're compatible with the upstream default helm chart installation. Previously, if you tried to install a vanilla jenkins installation we would bomb out for various plugin reasons.

### CODE REVIEW

**Comment by joshrwolf** [comment]
leaving in draft until https://github.com/chainguard-dev/apko/pull/1146 is landed

**Comment by found-it** [comment]
image

File: images/jenkins/config/template.apko.yaml [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
     - username: jenkins
-      uid: 65532
-  run-as: 65532
-
-work-dir: /app
+      gid: 1000
+      uid: 1000
+      homedir: "/var/jenkins_home"
+  # Match upstream
+  run-as: 1000
```
Comment (jamonation) [inline images/jenkins/config/template.apko.yaml:16]: Will this change be an issue for anyone using this with existing PVs that get remounted with new uid/gids?

Comment (joshrwolf) [reply images/jenkins/config/template.apko.yaml:16]: not if they're using the chart defaults, which were setting it to `1000` at runtime anyways

File: images/jenkins/config/template.apko.yaml [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
+      gid: 1000
   users:
     - username: jenkins
-      uid: 65532
-  run-as: 65532
-
-work-dir: /app
+      gid: 1000
+      uid: 1000
+      homedir: "/var/jenkins_home"
```
Comment (jamonation) [inline images/jenkins/config/template.apko.yaml:14]: CI and me both seem to be getting this message here? Should I be upgrading something?

```
│ yaml: unmarshal errors:
│   line 8: field homedir not found in type types.User
```

Comment (joshrwolf) [reply images/jenkins/config/template.apko.yaml:14]: working on rolling out all the necessary changes, this had to be added in apko https://github.com/chainguard-dev/apko/pull/1146

**Review by k4leung4 — APPROVED** [review]

**Review by jamonation — APPROVED** [review]
kubernetes credentials and everything, love it

**Pull request merged**